source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -530,7 +532,16 @@ void DefaultCertValidator::addClientValidationContext(SSL_CTX* ctx, bool require
   }
   // Set the verify_depth
   if (config_->maxVerifyDepth().has_value()) {
-    SSL_CTX_set_verify_depth(ctx, config_->maxVerifyDepth().value());
+    int max_verify_depth = std::min(config_->maxVerifyDepth().value(), uint32_t{INT_MAX});
```

Comment:
```suggestion
    uint32_t max_verify_depth = std::min(config_->maxVerifyDepth().value(), uint32_t{INT_MAX});
```

Reply:
That will cause the subtraction below to potentially overflow.